src/views/MissionPlanningView.vue

```diff
@@ -1515,8 +1708,20 @@ onMounted(async () => {
   await goHome()
   await nextTick()
 
-  planningMap.value.on('contextmenu', showContextMenu)
+  planningMap.value.on('contextmenu', (e: LeafletMouseEvent) => {
+    if (isCreatingSurvey.value) return
+    selectedWaypoint.value = undefined
+    if (selectedSurveyId.value === '') {
+      contextMenuType.value = 'map'
+      showContextMenu(e)
+    }
+    if (selectedSurveyId.value !== '') {
+      showContextMenu(e)
+    }
```

Comment:
This expression can be simplified since `showContextMenu(e)` will always be called in the end.